languages/typescript/docs/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,141 @@
-# How to implement an TypeScript concept exercise
+# How to implement a TypeScript Concept Exercise
 
-TODO: describe how to implement a concept exercise for the TypeScript track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes how to implement a Concept Exercise for the TypeScript track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<slug>`: the name of the exercise in snake_case (e.g. `anonymous-methods`).
+- `<concepts>`: the Concepts the exercise is about (e.g. `loops`),
+- `<concept-1>`: a single Concept slug,
+- `<prerequisite-n>`: a single Concept slug,
+- `<uuid>`: a _new_ v4 UUID (random!)
+- `<first-and-last-name>`: your first and last name (e.g. `John Doe`)
+- `<git-email>`: the email address you use for git (e.g. `johndoe@email.com`)
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Having done this, please read the [TypeScript Concept exercises introduction][concept-exercises]. If you have come up with something completely new, create a new issue _first_ so we can discuss the Concept Exercise.
+
+To implement a Concept Exercise, the following files must be created:
+
+<pre>
+languages
+└── typescript
+    └── exercises
+        └── concept
+            └── &lt;slug&gt;
+                ├── .docs
+                |   ├── instructions.md
+                |   ├── introduction.md
+                |   ├── hints.md
+                |   └── after.md (optional)
+                ├── .eslintrc
+                ├── .gitignore
+                ├── babel.config.js
+                ├── example.js
+                ├── &lt;slug&gt;.js
```

Comment:
Not `.js`, and probably `slug.test.ts`, `slug.ts` and `example.ts`?

Reply:
Good catch! I fixed those and updated the rest, i.e removed babel.config.js & .gitignore

Reply:
`.gitignore` _should_ probably be there 😄

Reply:
I was wondering if it would make more sense to have a single .gitignore in the root of the project that would take care of ignoring things across the repo instead of duplicating the file for every concept. I am assuming that a little regex in the gitignore should be able to make that happen.

Reply:
@SleeplessByte  I've added the .gitignore back to the folder structure 👍